### ISSUE TITLE
TensorFlow.TFException: No gradient defined for op: Pow.

### ISSUE THREAD

**Issue by tengge1** [migueldeicaza/TensorFlowSharp#187]
Hello, I use gradient descent method to do linear fit.

```C#
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using TensorFlow;

namespace LinearFit
{
    /// <summary>
    /// LinearFit
    /// </summary>
    class Program
    {
        static void Main(string[] args)
        {
            var xList = new List<double>();
            var yList = new List<double>();
            var ran = new Random();
            for (var i = 0; i < 10; i++)
            {
                var num = ran.NextDouble();
                var noise = ran.NextDouble();
                xList.Add(num);
                yList.Add(num * 3 + 4 + noise); // y = 3 * x + 4
            }
            var xData = xList.ToArray();
            var yData = yList.ToArray();
            var learning_rate = 0.01;

            var g = new TFGraph();

            var x = g.Placeholder(TFDataType.Double, new TFShape(xData.Length));
            var y = g.Placeholder(TFDataType.Double, new TFShape(yData.Length));

            var W = g.VariableV2(TFShape.Scalar, TFDataType.Double, operName: "weight");
            var b = g.VariableV2(TFShape.Scalar, TFDataType.Double, operName: "bias");

            var initW = g.Assign(W, g.Const(ran.NextDouble()));
            var initb = g.Assign(b, g.Const(ran.NextDouble()));

            var output = g.Add(g.Mul(x, W), b);

            var loss = g.ReduceSum(g.Pow(g.Sub(output, y), g.Const(2.0)));
            var grad = g.AddGradients(new TFOutput[] { loss }, new TFOutput[] { W, b });
            var optimize = new[]
            {
                g.AssignSub(W, g.Mul(grad[0], g.Const(learning_rate))).Operation,
                g.AssignSub(b, g.Mul(grad[1], g.Const(learning_rate))).Operation
            };

            var sess = new TFSession(g);

            sess.GetRunner().AddTarget(initW.Operation, initb.Operation).Run();

            for (var i = 0; i < 1000; i++)
            {
                var result = sess.GetRunner()
                    .AddInput(x, xData)
                    .AddInput(y, yData)
                    .AddTarget(optimize)
                    .Fetch(loss, W, b).Run();

                Console.WriteLine("loss: {0} W:{1} b:{2}", result[0].GetValue(), 
                    result[1].GetValue(), result[2].GetValue());
            }
        }
    }
}
```

It throw an exception when I call `AddGradients` method.

```
 No gradient defined for op: Pow. Please see 
https://www.tensorflow.org/code/tensorflow/cc/gradients/README.md 
for instructions on how to add C++ gradients.
```

How can I add gradient defined for op: Pow?

**Comment by tengge1** [comment]
It is ok when I change 
```
var loss = g.ReduceSum(g.Pow(g.Sub(output, y), g.Const(2.0)));
```
to
```
var loss = g.ReduceSum(g.Abs(g.Sub(output, y)));
```
. But I really want to use a gradient defined for op: Pow.

**Comment by cesarsouza** [comment]
Hi @tengge1,

Please submit this issue to the [TensorFlow issue tracker](https://github.com/tensorflow/tensorflow/issues). I've also [recently submitted one regarding the op.Select](https://github.com/tensorflow/tensorflow/issues/14845) that you can use as a base.

In short, the problem is happening because the gradients for ```op.Pow``` have not been implemented in TensorFlow's C++ API yet. TensorFlowSharp merely calls this API, so there is not much that can be done from TensorFlowSharp's side at the moment.

If you find any other operations that are missing gradients, please go submitting them to TensorFlow's issue tracker directly. The sooner we report them, the more likely they are going to be included in the next TensorFlow release. If we could manage to have most of them implemented in one go, it would be a real time saver.

Regards,
Cesar

**Comment by cesarsouza** [comment]
Hi @tengge1,

Thanks again for the reporting! I just wanted to let you know that I transferred the issue to the official TensorFlow issue tracker myself under https://github.com/tensorflow/tensorflow/issues/15239 as I am also interested in this issue.

Regards,
Cesar

**Comment by dgeller-OUHSC** [comment]
After failing with the same POW() error, I built TensorFlow from the latest master branch today and I was able to run the code above successfully,

**Comment by basejn** [comment]
Is this issue fixed in the official build.

**Comment by bendiy** [comment]
@migueldeicaza @cesarsouza @tengge1 
This issue is fixed now. The original example is running without error using the latest `master` of `TensorFlowSharp`. `Pow` is exposed in TensorFlow's C++ api.
